src/coord/coord.rs

```diff
@@ -1884,15 +1913,22 @@ where
         let mut uses_ids = Vec::new();
         source.global_uses(&mut uses_ids);
 
+        let need_to_determine = match when {
+            PeekWhen::Immediately => true,
+            _ => false,
+        };
```

Comment:
I think this could also be
```rust
let need_to_determine = when == PeekWhen::Immediately;
```
A match could help us avoid weird cases in the future if `PeekWhen` gains other variants, but the `_` match means that this probably isn't too different from the above, I think. I'd probably either switch to the above, or put an explicit match with the other `PeekWhen` variant so that the match explodes if we change the variants.

Reply:
I didn't do that just since it would have required adding an `Eq` derive to `PeekWhen`, but I can just do that, I can't see a reason not to.